crates/datasources/src/bigquery/mod.rs

```diff
@@ -476,37 +477,55 @@ fn bigquery_table_to_arrow_schema(table: &Table) -> Result<ArrowSchema> {
-            FieldType::String => DataType::Utf8,
-            FieldType::Integer | FieldType::Int64 => DataType::Int64,
-            FieldType::Float | FieldType::Float64 => DataType::Float64,
-            FieldType::Bytes => DataType::Binary,
-            FieldType::Date => DataType::Date32,
-            // BigQuery actually returns times with microsecond precision. We
-            // aim to work only with nanoseconds to have uniformity accross the
-            // codebase. It's also easier to have interop with datafusion since
-            // with many things like type inference datafusion uses nanosecond.
-            // This cast is done when the stream is received by using the
-            // `crate::common::util::normalize_batch` function.
-            FieldType::Datetime => DataType::Timestamp(TimeUnit::Nanosecond, None),
-            FieldType::Timestamp => DataType::Timestamp(TimeUnit::Nanosecond, Some("UTC".into())),
-            FieldType::Time => DataType::Time64(TimeUnit::Nanosecond),
-            FieldType::Numeric => DataType::Decimal128(38, 9),
-            FieldType::Geography => DataType::Utf8,
-            other => return Err(BigQueryError::UnsupportedBigQueryType(other.clone())),
-        };
-
-        let field = Field::new(&field.name, arrow_typ, true);
-        arrow_fields.push(field);
+        let resolved_field = table_field_schema_to_arrow_datatype(field);
+        if let Ok(resolved_field) = resolved_field {
+            arrow_fields.push(resolved_field);
+        }
     }
-
     Ok(ArrowSchema::new(arrow_fields))
 }
 
+fn table_field_schema_to_arrow_datatype(field: &BigQuerySchema) -> Result<Field> {
+    // See <https://cloud.google.com/bigquery/docs/reference/storage#arrow_schema_details>
+    // for how BigQuery types map to Arrow types.
+    let arrow_typ = match &field.r#type {
+        FieldType::Bool | FieldType::Boolean => DataType::Boolean,
+        FieldType::String => DataType::Utf8,
+        FieldType::Integer | FieldType::Int64 => DataType::Int64,
+        FieldType::Float | FieldType::Float64 => DataType::Float64,
+        FieldType::Bytes => DataType::Binary,
+        FieldType::Date => DataType::Date32,
+        // BigQuery actually returns times with microsecond precision. We
+        // aim to work only with nanoseconds to have uniformity accross the
+        // codebase. It's also easier to have interop with datafusion since
+        // with many things like type inference datafusion uses nanosecond.
+        // This cast is done when the stream is received by using the
+        // `crate::common::util::normalize_batch` function.
+        FieldType::Datetime => DataType::Timestamp(TimeUnit::Nanosecond, None),
+        FieldType::Timestamp => DataType::Timestamp(TimeUnit::Nanosecond, Some("UTC".into())),
+        FieldType::Time => DataType::Time64(TimeUnit::Nanosecond),
+        FieldType::Numeric => DataType::Decimal128(38, 9),
+        FieldType::Bignumeric => DataType::Decimal256(76, 9),
```

Comment:
This was intentionally omitted, but it's okay. Using big numeric will still cause errors since we don't know how to process these decimals.

Reply:
Also, Is it for sure `(76, 9)` ?

Reply:
/// The maximum precision for [DataType::Decimal128] values
pub const DECIMAL128_MAX_PRECISION: u8 = 38;

/// The maximum scale for [DataType::Decimal128] values
pub const DECIMAL128_MAX_SCALE: i8 = 38;

/// The maximum precision for [DataType::Decimal256] values
pub const DECIMAL256_MAX_PRECISION: u8 = 76;

/// The maximum scale for [DataType::Decimal256] values
pub const DECIMAL256_MAX_SCALE: i8 = 76;

Reply:
Oh yeah, I meant is the default scale 9 here as well?

Reply:
default scale is 10 for both Decimal128 and Decimal256

Reply:
https://cloud.google.com/bigquery/docs/reference/storage#arrow_schema_details

It's 38 for big numeric